bin/index.js

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env node
+var DraftLog = require('draftlog').into(console).addLineListener(process.stdin);
+var chalk = require('chalk');
+var SpeedTestNet = require('../');
+
+/*
+ * Keep track of current UI State
+ */
+var header, speeds, location
+var step = 'Ping';
+var statuses = {
+  Ping: true,
+  Download: false,
+  Upload: false,
+}
+
+/*
+ * Renders the header and Speeds (one below the other)
+ */
+var halfwidth = 12;
+function updateLines() {
+  var headerTxt = '';
+  var speedsTxt = '';
+  var spinner = Spinner();
+
+  // Build Header
+  for (var k in statuses) {
+    var status = statuses[k];
+
+    // Create header
+    var spacing = ' '.repeat(halfwidth - k.length / 2);
+    var col = spacing + k + spacing;
+
+    // Set header status color
+    if (status === false) {
+      col = chalk.dim(col);
+    } else {
+      col = chalk.white(col);
+    }
+
+    headerTxt += col;
+  }
+
+  // Build Status
+  for (var k in statuses) {
+    var status = statuses[k];
+    status = status === false ? '' : status;
+    status = status === true  ? '' : status;
+    status = status + '';
+
+    // Pad to be even
+    status = (status.length % 2 == 0 ? '' : ' ') + status;
+
+    // Put spinner if no info
+    if (!status) {
+      status = spinner + ' ';
+    }
+
+    // Create status text
+    var spacing = ' '.repeat(halfwidth - status.length / 2);
+
+    // If it's the current step, show as yellow
+    if (step == k) {
+      var txt = spacing + chalk.yellow(status) + spacing;
+    } else {
+      var txt = spacing + chalk.blue(status) + spacing;
+    }
+
+    speedsTxt += txt;
+  }
+
+  header('│' + headerTxt + '│');
+  speeds('│' + speedsTxt + '│');
+}
+
+/*
+ * Function that return state of Spinner, and change its state with time
+ */
+var frames = ['⠁', '⠂', '⠄', '⡀', '⢀', '⠠', '⠐', '⠈' ];
+var lastChange = 0;
+function Spinner(){
+  if (Date.now() > lastChange + 0) {
+    frames.unshift(frames.pop());
+    lastChange = Date.now();
+  }
+  return frames[0];
+}
+
+/*
+ * Shows CLI UI
+ */
+console.log();
+console.log('┌' + '─'.repeat(halfwidth * 6) + '┐');
+console.log('│' + ' '.repeat(halfwidth * 6) + '│');
+location = console.draft('│' + ' '.repeat(halfwidth * 6) + '│');
+console.log('│' + ' '.repeat(halfwidth * 6) + '│');
+
+header = console.draft();
+speeds = console.draft();
+
+updateLines();
+
+console.log('│' + ' '.repeat(halfwidth * 6) + '│');
+console.log('│' + ' '.repeat(halfwidth * 6) + '│');
+console.log('└' + '─'.repeat(halfwidth * 6) + '┘');
+console.log();
+console.log();
+
+/*
+ * Start speed test
+ */
+var test = SpeedTestNet({maxTime: 10000});
+var interval = setInterval(updateLines, 60);
```

Comment:
The `interval` variable isn't used anywhere so it's probably best if we get rid of it. Also, could you increase the interval to 100ms? It makes the spinner less frantic and the console more stable (less blinking).